packages/page-settings/src/modals/NetworkSpecs.tsx

```diff
@@ -0,0 +1,180 @@
+      setNetworkSpecs(defaultNetworkSpecs);
+    };
+    getNetworkSpec();
+  }, [api]);
+
+  type inputListener = (v: string) => void;
+  const _onChangeValue = (k: keyof NetworkSpecsStruct): inputListener => (v: string): void => setNetworkSpecs({ [k]: v });
+  const _onSetRandomColor = (): void => setNetworkSpecs({ color: getRandomColor() });
+  const _checkColorValid = (): boolean => /^#[\da-fA-F]{6}|#[\da-fA-F]{3}$/.test(networkSpecs.color);
+
+  return <Modal
+    className={className}
+    header={t('Export Network Specs')}
+  >
+    <Modal.Content>
+      <Input
+        autoFocus
+        className='full'
+        help={t('Name of the network. It only for display purpose.')}
+        label={t('Network Name')}
+        onChange={_onChangeValue('title')}
+        value={networkSpecs.title}
+      />
+      <Input
+        className='full'
+        help={t('The color used to distinguish this network with others, use color code with 3 or 6 digits, like "#FFF" or "#111111"')}
+        isError={!_checkColorValid()}
+        label={t('Color')}
+        onChange={_onChangeValue('color')}
+        value={networkSpecs.color}
+      >
+        <div className='settings--networkSpecs-colorButton'>
+          <Button
+            label={t('Generate Random Color')}
+            icon='sync'
+            key='spread'
+            onClick={_onSetRandomColor}
+          />
+          <ColoredComponent color={networkSpecs.color}/>
+        </div>
+      </Input>
+      <Input
+        className='full'
+        help={t('Genesis Hash refers to initial state of the chain, it cannot be changed once the chain is launched')}
+        isDisabled
+        label={t('Genesis Hash')}
+        value={networkSpecs.genesisHash}
+      />
+      <Input
+        className='full'
+        help={t('Unit decides the name of 1 unit token, e.g. "DOT" for Polkadot')}
+        isDisabled
+        label={t('Unit')}
+        value={networkSpecs.unit}
+      />
+      <Input
+        className='full'
+        help={t('Prefix indicates the ss58 address format in this network, is a number between 0 ~ 255 describes the precise format of the bytes of the address')}
+        isDisabled
+        label={t('Address Prefix')}
+        value={networkSpecs.prefix}
+      />
+      <Input
+        className='full'
+        help={t('Decimals decides the smallest unit of the token, which is 1/10^decimals')}
+        isDisabled
+        label={t('Decimals')}
+        value={networkSpecs.decimals}
+      />
+      <QrNetworkSpecs className='settings--networkSpecs-qr' networkSpecs={debouncedQrData}/>
+    </Modal.Content>
+    <Modal.Actions onCancel={onClose} withOr={false}>
+    </Modal.Actions>
+  </Modal>;
+}
+
+interface ColoredComponentProps extends BareProps {
+  color: string;
+}
+const ColoredComponent = React.memo(styled(({ color, className }: ColoredComponentProps) => (<div color={color} className={className}/>))`
```

Comment:
Prefer not doing this. Just add a class on main.

Reply:
what do you mean `on main`?

Reply:
The styled components end up being less than optiomal. Have to deal with that in the extension, here have managed to keep it (somewhat) into control. So there is nothing wrong with -

```
<div className={className}>
  ...
  <div className='sub'>...</div>
</div>

export React.memo(styled(Main)`
...
.sub { ... }
```

Reply:
I mean by `on main` where to put this component? what is the legacy way to do that?

Also NetworkSpecs.tsx need to become a component now. 

 My idea to create an `extension` folder, and put this `ColoredComponent.tsx` together with `NetworkSpecs.tsx` into it, and split `Extension.tsx` into  two parts : `index.tsx` and `ExtensionInjection`.